src/services/text_processor.py

```
@@ -1,39 +1,37 @@
-                )
-            except OSError:
-                raise TextProcessingError(
-                    "No spaCy model found. Please install with: python -m spacy download en_core_web_md"
-                )
+            self.sentence_transformer = SentenceTransformer(self.bert_model_name)
+            print(f"Loaded sentence transformer: {self.bert_model_name}")
+        except Exception as e:
+            raise TextProcessingError(
+                f"Could not load sentence transformer model '{self.bert_model_name}'. "
+                f"Error: {e}. Please install with: pip install sentence-transformers"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)